tasks/config/run.js

```diff
@@ -177,6 +147,14 @@ module.exports = function (grunt) {
         'scripts/notice',
         '--validate'
       ]
-    }
+    },
+
+    api_integration_tests: {
+      cmd: process.execPath,
+      args: [
+        'scripts/functional_tests',
+        '--config ../../test/api_integration/jenkins_config.js',
+      ],
+    },
```

Comment:
* Remove `test:api` related code.
* Replace it with a call to `@kbn/test` code, scoped to just run api tests.
* Run api_integration_tests using the `scripts/functional_tests` (which uses `@kbn/test`
    * Pass in just the config for api_integration tests
    * `jenkins_config.js` is built on the original with just a change to run ES from `source`

Alternative way to do this:
* Call `runTests` directly from `@kbn/test`, though I'm not sure how to do this. If it needs to be done in a command-line way, then we already have it in `scripts/functional_tests`.
* Allow `test/api_integration/config.js` to take in a `from`.
* Or allow `scripts/functional_tests` to take in a `--from`.
    * This seems worth trying to compare.